src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(args) {
+    if (args === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
```

Comment:
when we're returning both in if and else you can omit else statement, it will be reached eventually 😄